### DISCUSSION TITLE
weird naming convention and inconsistency for losing items and skills on server configs

### DISCUSSION THREAD

**Discussion by ghost** [SmartlyDressedGames/Unturned-3.x-Community#4461] [Ideas]
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/67235885/9a7f6662-14ef-4f7d-a346-d5e73799c634)

skills and experience are based on how much you KEEP from 0% to 100%, items are based on how much you LOSE, from 0% to 100%
this has been the cause of confusion to some people ive spoken to (including myself), we find it odd that these two terms mean different variables.

i propose swapping around the values for the lose_items parameters, so that 1.0 keeps all your items, and 0.0 loses them, like how skills and experience works.

renaming "Lose" to "Keep" would probably work better too, as "lose 0%" meaning you lose everything, doesnt really make sense?

**Comment by SDGNelson** [comment]
I agree, that's a much better naming for it! Unturned has a lot of bad naming baggage.

Although using automatic json serialization for the config makes it easy to add new properties quickly, one downside is it's trickier to handle automatic conversion and versioning of properties. There is probably a way to do it with Json.NET but I haven't dug into it much. As many downsides as there are to the .dat format for assets, I have been really happy with how easy it is to handle backwards compatibility and versioning. I can rename properties or change how something is parsed while keeping some code around to convert old properties.

The reason I mention that is at some point I would like to rewrite the server config to use our .dat files. We could go through and rename some properties, better handle versioning, get rid of the easy/normal/hard section split, etc.

**Comment by ghost** [comment]
i think that easy/normal/hard should just be setting presets like how server filters have presets like internet or somethin
cus you can change all the settings on hard to be the same as easy, for example. so i think the buttons should just set the settings yknow

**Reply by ghost** [reply]
1 sec lemme visualise what i mean

**Comment by ghost** [comment]
ok i think it would be nice if we had the presets system from the server browser but for the advanced configs
i know we are mainly on about server stuff but i think this would be cool
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/67235885/14f966e7-8da5-45f4-a639-4b9140b46208)
excuse my wonderful paintnet skills but i think i got the point accross yknow
